src/events/types/inventory/InventoryEvent.ts

```diff
@@ -0,0 +1,27 @@
+import { Player } from '../../../player/Player';
+import { Inventory } from '../../../inventory/Inventory';
+import { InventoryView } from '../../../inventory/InventoryView';
+import { Event } from '../Event';
+
+export class InventoryEvent extends Event {
+  /**
+   * Gets the primary Inventory involved in this transaction
+   */
+  public getInventory(): Inventory {
+    return Inventory.fromJava(this.toJava().getInventory());
+  }
+
+  /**
+   * Gets the list of players viewing the primary (upper) inventory involved in this event
+   */
+  public getViewers(): readonly Player[] {
+    return Object.freeze(this.toJava().getViewers().map((entity: any) => Player.fromJava(entity)));
+  }
+
+  /**
+   * Gets the view object itself
+   */
+  public getView(): InventoryView {
+    return new InventoryView(this.toJava().getView());
```

Comment:
Don't worry about it now, but eventually I'll want to make an `InventoryView.fromJava` function instead of using the constructor.

Reply:
Got it, this can be eddited when it's added